comments-bundle/src/EventListener/DataContainer/CommentsConditionalFieldsListener.php

```diff
@@ -0,0 +1,176 @@
+            'sql' => ['type' => 'boolean', 'default' => false],
+        ];
+
+        switch ($table) {
+            case 'tl_news':
+            case 'tl_calendar_events':
+                PaletteManipulator::create()
+                    ->addLegend('comments_legend', 'protected_legend', PaletteManipulator::POSITION_AFTER, true)
+                    ->addField(['allowComments'], 'comments_legend', PaletteManipulator::POSITION_APPEND)
+                    ->applyToPalette('default', $table)
+                ;
+
+                break;
+
+            case 'tl_faq':
+                PaletteManipulator::create()
+                    ->addLegend('comments_legend', 'title_legend', PaletteManipulator::POSITION_AFTER, true)
+                    ->addField(['allowComments'], 'comments_legend', PaletteManipulator::POSITION_APPEND)
+                    ->applyToPalette('default', $table)
+                ;
+
+                break;
+        }
+    }
+
+    private function applyChildrenFields(string $table): void
+    {
+        $GLOBALS['TL_DCA'][$table]['list']['sorting']['headerFields'][] = 'allowComments';
+
+        switch ($table) {
+            case 'tl_news':
+                $GLOBALS['TL_DCA'][$table]['fields']['noComments'] = [
+                    'filter' => true,
+                    'inputType' => 'checkbox',
+                    'eval' => ['tl_class' => 'w50 m12'],
+                    'sql' => ['type' => 'boolean', 'default' => false],
+                ];
+                break;
+
+            case 'tl_calendar_events':
+                $GLOBALS['TL_DCA'][$table]['fields']['noComments'] = [
+                    'inputType' => 'checkbox',
+                    'eval' => ['tl_class' => 'w50 m12'],
+                    'sql' => ['type' => 'boolean', 'default' => false],
+                ];
+                break;
+
+            case 'tl_faq':
+                $GLOBALS['TL_DCA'][$table]['fields']['noComments'] = [
+                    'filter' => true,
+                    'inputType' => 'checkbox',
+                    'sql' => ['type' => 'boolean', 'default' => false],
+                ];
+                break;
+        }
+
+        switch ($table) {
+            case 'tl_news':
+            case 'tl_calendar_events':
+                PaletteManipulator::create()
+                    ->addField(['noComments'], 'expert_legend', PaletteManipulator::POSITION_APPEND)
+                    ->applyToPalette('default', $table)
+                    ->applyToPalette('internal', $table)
+                    ->applyToPalette('article', $table)
+                    ->applyToPalette('external', $table)
+                ;
+
+                break;
+
+            case 'tl_faq':
+                PaletteManipulator::create()
+                    ->addLegend('expert_legend', 'publish_legend', PaletteManipulator::POSITION_BEFORE, true)
+                    ->addField(['noComments'], 'expert_legend', PaletteManipulator::POSITION_APPEND)
```

Comment:
```suggestion
                    ->addField('noComments', 'expert_legend', PaletteManipulator::POSITION_APPEND)
```

Just a minor nitpick and could also stay as-is. Applies to other places as well.